### DISCUSSION TITLE
missing C++ header

### DISCUSSION THREAD

**Discussion by John4650-hub** [termux/termux-packages#22288] [Q&A] [answered]
I am building a new package which uses  nlohmann's json but can't get past this error.
I installed the header from the workflow file 
using the following command 
```bash
sudo apt-get install nlohmann-json3-dev
```
Also iam using docker and have been developing from github actions. I don't know what  to do .
Here's how i includer the header
```cpp
#include <iostream>
#include <fstream>
#include <vector>
#include <string>
#include <sstream>
#include <filesystem>
#include <unordered_map>
#include <nlohmann/json.hpp>

using namespace std;
namespace fs = filesystem;
using json = nlohmann::json;
```
```text
fatal error: 'nlohmann/json.hpp' file not found
```

**Comment by TomJo2000** [comment, marked as answer]
Termux does not split out headers into separate `*-dev` packages.
This header should be part of the standard `nlohmann-json` package.

In fact it's right where it should be in the package.
![image](https://github.com/user-attachments/assets/d23fa6a0-248f-41c1-b931-bf597c2cacb0)

**Reply by John4650-hub** [reply]
@TomJo2000  could you tell me why am getting the error

**Reply by Biswa96** [reply]
apt exists.

```
$ apt search nlohmann
Sorting... Done
Full Text Search... Done
nlohmann-json/stable 3.11.3 aarch64
  JSON for Modern C++
```

**Reply by TomJo2000** [reply]
You'll need to ensure that the necessary dependencies are available inside your docker environment.
I do not know what sort of setup you are using here, but this is not an issue with the Termux package of `nlohmann-json`.
